### ISSUE TITLE
closure-compiler-unshaded v20240317 embeds jakarta.annotation-api

### ISSUE THREAD

**Issue by tmortagne** [google/closure-compiler#4161]
It can cause some conflicts when you depend on both `closure-compiler-unshaded` and `jakarta.annotation-api` directly or through other dependencies.

I would have expected it to be a dependency, along with the others declared in the pom file.

**Comment by brad4d** [comment]
Thanks for pointing this out.
We're unlikely to do anything about it immediately,
but I expect we will fix it before doing another release.

**Comment by tmortagne** [comment]
Thanks @brad4d, that's definitely good enough for me 🙂

**Comment by gkdn** [comment]
@tmortagne Do you know if this has regressed in the latest release?

**Comment by tmortagne** [comment]
Yes, as detailed by @niloc132 and as I should have mentioned a bit more explicitly in the issue description, it's indeed a regression from v20231112.

**Comment by gkdn** [comment]
https://github.com/google/closure-compiler/commit/4bf1cb0841d4f88a25ecd6ba66615e861c799aae will fix the issue in the next release.

**Comment by mflorea** [comment]
Are there any plans for a new release to fix this issue? It's been one year since it was reported... Maybe you just need to publish the released artifacts on Maven Central. I see 7 published releases for 2023, only one for 2024 and none for 2025 so far on https://central.sonatype.com/artifact/com.google.javascript/closure-compiler-unshaded . Thanks.

**Comment by brad4d** [comment]
@lauraharker implemented the creation of the tags for each of our internal-to-Google releases.
@jDramaix is working on automation that will use those tags for the next and future releases to Maven.

**Comment by lauraharker** [comment]
Update: we've pushed a new Maven release - v20250402.

**Comment by tmortagne** [comment]
Thanks, @lauraharker. Unfortunately, we cannot use it yet because of the new dependency on Java 21.
